No assertion rewriting in files imported from conftest

I guess I can't be the only one importing some utility code from `conftest.py`, or can I? :wink:

With this `conftest.py`:

```python
import pytest
from testutils import SomeClass

@pytest.fixture
def some_obj():
    return SomeClass()
```

And a `testutils.py`:

```python
class SomeClass():

    def __init__(self):
        x = 23
        y = 42
        assert x + y == 1337
```

And a `test_foo.py`:

```python
def test_foo(some_obj):
    pass
```

With pytest 2.9, I guess assertion reinterpretion kicks in and I get:

```
>       assert x + y == 1337
E       AssertionError
```

With 3.0, that's removed and I get plain asserts:

```
>       assert x + y == 1337
E       AssertionError
```

Since I have some custom assertion helpers and stuff in separated files, that's quite a bummer.

I'm not sure what the best way to fix this would be - ideas I had so far:

- Rewrite everything under the testpath if an explicit one is given
- Rewrite everything in directories where there are `conftest.py` files
- Rewrite everything imported from `conftest.py` files (probably cleanest, but maybe hard to implement?)

You will have to call `register_assert_rewrite` before trying to import `testutils`:

```python
import pytest
pytest.register_assert_rewrite('testutils')
from testutils import SomeClass

@pytest.fixture
def some_obj():
    return SomeClass()
```

This now works as expected:

```
    def __init__(self):
        x = 23
        y = 42
>       assert x + y == 1337
E       assert (23 + 42) == 1337
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -89,7 +89,11 @@ time or change existing behaviors in order to make them less surprising/more use
 
 * Reinterpretation mode has now been removed.  Only plain and rewrite
   mode are available, consequently the ``--assert=reinterp`` option is
-  no longer available.  Thanks `@flub`_ for the PR.
+  no longer available.  This also means files imported from plugins or
+  ``conftest.py`` will not benefit from improved assertions by
+  default, you should use ``pytest.register_assert_rewrite()`` to
+  explicitly turn on assertion rewriting for those files.  Thanks
+  `@flub`_ for the PR.
 
 * The following deprecated commandline options were removed:
 

